### DISCUSSION TITLE
Bug inline IF and Else statemant inside wire:poll

### DISCUSSION THREAD

**Discussion by ghost** [livewire/livewire#4385] [1. Help] [answered]
When use inline if and else statement inside wire:poll, if else not working,

please fix it -thankyou

**Comment by joshhanley** [comment]
@brianalviano thanks for reporting! Could you share your code which reproduces the issue with as little code as possible so we can investigate it?

Thanks!

**Reply by ghost** [reply]
![Screenshot 2021-12-14 132920](https://user-images.githubusercontent.com/36157055/145945117-772a11af-b1ae-48e5-bf5d-6c34c8e370db.png)

**Reply by zepfietje** [reply]
@brianalviano this is not a bug with `wire:poll`. What happens here is that when the component polls, Livewire sends a request to the server. This is a request to a route Livewire uses internally to send messages back and forth. That's why `Request::is()` doesn't work at that point (it basically only works during the first request when mounting a Livewire component.

How to fix this issue? You can create a property on your Livewire component which stores the current route name for example (as I assume you want to perform the active check for multiple navigation links). If you set this value in the component mount method, everything should work fine. Try something like this:
```php
public string $routeName;

public function mount(): void
{
    $this->routeName = Route::currentRouteName();
}
```
```blade.php
<a class="nav-main-link {{ $routeName === 'peneliti.dashboard' ? 'active' : '' }}>
```

**Reply by joshhanley** [reply]
@brianalviano yeah what @zepfietje said is correct 🙂 @zepfietje do you want to copy your comment to an answer below, so it can be accepted as an answer?

**Reply by zepfietje** [reply]
@joshhanley, done! ✅

**Reply by joshhanley** [reply]
Thanks! 🙂

**Comment by zepfietje** [comment, marked as answer]
@brianalviano this is not a bug with `wire:poll`. What happens here is that when the component polls, Livewire sends a request to the server. This is a request to a route Livewire uses internally to send messages back and forth. That's why `Request::is()` doesn't work at that point (it basically only works during the first request when mounting a Livewire component.

How to fix this issue? You can create a property on your Livewire component which stores the current route name for example (as I assume you want to perform the active check for multiple navigation links). If you set this value in the component mount method, everything should work fine. Try something like this:
```php
public string $routeName;

public function mount(): void
{
    $this->routeName = Route::currentRouteName();
}
```
```blade.php
<a class="nav-main-link {{ $routeName === 'peneliti.dashboard' ? 'active' : '' }}>
```
